controller/rockcraft.yaml

```diff
@@ -0,0 +1,61 @@
+# Dockerfile https://github.com/kserve/kserve/blob/v0.11.2/Dockerfile
+name: kserve-controller
+summary: KServe controller
+description: "KServe controller manager"
+version: "0.11.2"
+license: Apache-2.0
+base: ubuntu@22.04
+platforms:
+    amd64:
+run-user: _daemon_
+
+services:
+  kserve-controller:
+    override: replace
+    summary: "KServe controller manager service"
+    startup: enabled
+    command: "/manager"
+
+parts:
+  security-team-requirement:
+    plugin: nil
+    override-build: |
+      mkdir -p ${CRAFT_PART_INSTALL}/usr/share/rocks
+      (echo "# os-release" && cat /etc/os-release && echo "# dpkg-query" && \
+      dpkg-query --root=${CRAFT_PROJECT_DIR}/../bundles/ubuntu-22.04/rootfs/ -f '${db:Status-Abbrev},${binary:Package},${Version},${source:Package},${Source:Version}\n' -W) \
+      > ${CRAFT_PART_INSTALL}/usr/share/rocks/dpkg.query
+
+  controller:
+    plugin: go
+    source: https://github.com/kserve/kserve
+    source-type: git
+    source-tag: v0.11.2
+    build-snaps:
+      - go/1.20/stable
+    build-environment:
+      - CGO_ENABLED: 0
+      - GOOS: linux
+    override-build: |
+
+      # Remove the source from the build directory so we don't
+      # accidentally use the wrong inputs.
+      # The build steps below copy everything that is needed to the working directory
+      rm -rf ./*
```

Comment:
To better understand this decision, what are the wrong inputs and why are they taken automatically?

It feels a bit odd to just remove everything from the wd, maybe creating a temporal directory and working from there could help?

Reply:
commands we are running in this rock use files from the working directory as input, for example the `go mod download` command fetches the `go.mod` and `go.sum` files. Rockcraft by default copies the src to the build directory (the `$CRAFT_PART_SRC`), so we are removing the src in this step and copying only what we need. Your suggestion to create a temp dir and copying the needed files to it would have the same effect I think, but is there any reason we want to keep the rest of the files in the build directory?

Reply:
This is a style decision that was used by Andrew first IIRC in places like [here](https://github.com/canonical/katib-rocks/blob/5c919e8ff0346146e8a4d458474164e405a3e7ba/katib-ui-rock/rockcraft.yaml#L37). I also had questions on this when I first saw this and we had discussed it [here](https://github.com/canonical/katib-rocks/pull/38#discussion_r1409596822). We 've taken a note of this in our [how do you rock notes](https://docs.google.com/document/d/1Zm11x2sRxue3sUEWe776-6ZHPwyDl6hVXP41z2zhCno/edit).

Reply:
Thanks for the context @orfeas-k ! The comment above the `rm` command makes it really hard to understand what's actually going on. I think we should change it to reflect that we are essentially emptying the `build` directory and copying the build files that are specified in the upstream `Dockerfile`.

I don't think this hurt us in any way, but it is definitely not intuitive. To keep consistency with other rocks, I think we could be fine keeping the `rm` command and just copying the necessary files, but I would've created a tmp workingdir and do all things from there if in the end we are essentially just copying files around.

Reply:
+1 to making the comment more clear, we can also discuss this in the how do we rock meeting